### PR TITLE
Replace metalk8s-loki storageclass with metalk8s one

### LINKED ISSUE #2742: Replace various metalk8s-* storageclass by a single metalk8s one
**Component**: salt, addons, docs

**Why this is needed**:
Today we have 1 metalk8s-prometheus (prometheus & alertmanager) and 1 metalk8s-loki (loki & fluent-bit) storageclass, but it does not make sense as the storageclass name must not be used to define the purpose of it.
Moreover, these class are exactly the same, so there is no point of keeping both.

**What should be done**:
Replace `metalk8s-prometheus` and `metalk8s-loki` storageclass everywhere by a `metalk8s` one.

**Implementation proposal** (strongly recommended):
Should be updated in examples, docs and salt

**Test plan**:
Tests should still work.

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: salt, docs, charts

**Context**: See #2742

**Summary**:
* Update Loki helm chart to support selector for PVC (also opened an upstream PR https://github.com/grafana/loki/pull/2554)
* Replace the `metalk8s-loki` storageclass with `metalk8s` one
* Raise retries for logging tests.

**Acceptance criteria**: 

---

Closes: #2742

### CODE REVIEW

File: charts/loki/templates/statefulset.yaml [hunk 1]
```diff
@@ -124,5 +124,9 @@ spec:
         requests:
           storage: {{ .Values.persistence.size | quote }}
       storageClassName: {{ .Values.persistence.storageClassName }}
+      {{- if .Values.persistence.selector }}
```
Comment (Ebaneck) [inline charts/loki/templates/statefulset.yaml:127]: Hopefully, we don't forget this local patch during Loki chart upgrades...

**Review by Ebaneck — APPROVED** [review]
LGTM

File: charts/loki/templates/statefulset.yaml (hunk 1 above)
Comment (alexandre-allard) [reply charts/loki/templates/statefulset.yaml:127]: That's why I opened an upstream PR, I hope it'll be merged.

Comment (Ebaneck) [reply charts/loki/templates/statefulset.yaml:127]: Ha, I see the commit message now :+1:

**Pull request merged**